Release/src/http/client/x509_cert_utilities_apple.cpp

```diff
@@ -99,28 +99,58 @@ bool verify_X509_cert_chain(const std::vector<std::string> &certChain, const std
     // Note: SecTrustCreateWithCertificates expects the certificate to be
     // verified is the first element.
     cf_ref<CFStringRef> cfHostName = CFStringCreateWithCStringNoCopy(kCFAllocatorDefault,
-                                                                    hostName.c_str(),
-                                                                    kCFStringEncodingASCII,
-                                                                    kCFAllocatorNull);
-    if(cfHostName.get() == nullptr)
+        hostName.c_str(),
+        kCFStringEncodingASCII,
+        kCFAllocatorNull);
+    if (cfHostName.get() == nullptr)
     {
         return false;
     }
+
     cf_ref<SecPolicyRef> policy = SecPolicyCreateSSL(true /* client side */, cfHostName.get());
     cf_ref<SecTrustRef> trust;
     OSStatus status = SecTrustCreateWithCertificates(certsArray.get(), policy.get(), &trust.get());
-    if(status == noErr)
+
+    bool isVerified = false;
+    if (status == noErr)
     {
         // Perform actual certificate verification.
         SecTrustResultType trustResult;
         status = SecTrustEvaluate(trust.get(), &trustResult);
-        if(status == noErr && (trustResult == kSecTrustResultUnspecified || trustResult == kSecTrustResultProceed))
+        if (status == noErr && (trustResult == kSecTrustResultUnspecified || trustResult == kSecTrustResultProceed))
         {
-            return true;
+            isVerified = true;
         }
-    }
 
-    return false;
+        if (certInfoFunc)
+        {
+            auto info = std::make_shared<certificate_info>(hostName);
+            info->certificate_error = (long)trustResult;
+            info->verified = isVerified;
+
+            CFIndex cnt = SecTrustGetCertificateCount(trust.get());
```

Comment:
You could extract the following into a helper method to 'build certificate info'. 

We're adding quite a bit to this method